### ISSUE TITLE
[Nightly] Crash when selecting setting "Navigation menu" 

### ISSUE THREAD

**Issue by MagpieFourtyTwo** [cgeo/cgeo#13976]
### Describe your problem!

While trying to give Cruiser a try, I also tapped "Settings" => "Routing / Navigation" => "Navigation menu" for the first time since I guess years ... and it immediately crashed c:geo. 

### How to reproduce?

- open "Settings" 
- open "Routing / Navigation"
- tap "Navigation menu"

### Actual result after these steps?

Crash, c:geo restarts with main screen

### Expected result after these steps?

Just guess ... ;)

### Reproducible

Yes

### c:geo Version

2023.02.12-NB-6418ca

### System information

```text
## System information

c:geo version: 2023.02.12-NB-6418ca6

Device:
-------
- Device type: VOG-L29 (VOG-L29EEA, HUAWEI)
- Available processors: 8
- Android version: 10
- Android build: VOG-L29 10.1.0.150(C431E19R2P5)
- Screen resolution: 1080x2147px (423x841dp)
- Pixel density: 2.55
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: disabled - 23.04.13 (100400-505809224)

Sensor and location:
-------
- Low power mode: active
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:342
- Set language: en_DE (system default)
- System date format: dd/MM/y
- Time zone: GMT+01:00
- Debug mode active: no
- Last backup: 10 Feb, 22:10
- Routing mode: Car
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Map: Combined (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMultiMapSource
  - Atts: _ DE BaWü Vosges.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE Bayern.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE Hessen.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE North.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE NRW.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE Rheinland-Pfalz.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / _ DE Saarland Lux.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / AT CH Alps.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / BE LUX Belgium Lux.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / ES Spain Portugal.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / FR North.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / FR South.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / IT Italy_oam.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / NL Netherlands.osm.map: / VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2022 /  / World.map: / mapsforge-map-writer-dev-SNAPSHOT
  - Theme: Elevate5_Locus/Elevate_Locus.zip:Elevate.xml

Filters:
-------
- Hide waypoints: visited
- LIVE: None ([:inconclusive=false:advanced=true])
- OFFLINE: None ([:inconclusive=false:advanced=true])

Stored Filters (#9):
- -  Live (Hide Found, Own & Archived): [-  Live \(Hide Found, Own & Archived\):inconclusive=false:advanced=true]AND(status:owned_no:found_no:has_offline_found_log_no:exclude_archived))
- - HQ + interessantes Zeug: [- HQ + interessantes Zeug:inconclusive=false:advanced=true]AND(status:found_no:has_offline_found_log_no;stored_list:162:356:36:358:299:365:45:366:371:245:376:377:314:250:381:384:321:385:129:258:68:337:210:148:344:283:155:220:349:352))
- - Stored in any list: [- Stored in any list:inconclusive=false:advanced=true]AND(status:owned_no:found_no:stored_yes:has_offline_found_log_no:exclude_archived))
- - Stored in any list (no climbing): [- Stored in any list \(no climbing\):inconclusive=false:advanced=true]AND(attributes:treeclimbing_yes:inverse=true:attributesources=3;attributes:hike_long_yes:inverse=true:attributesources=3;attributes:swimming_yes:inverse=true:attributesources=3;attributes:wading_yes:inverse=true:attributesources=3;attributes:boat_yes:inverse=true:attributesources=3;attributes:rappelling_yes:inverse=true:attributesources=3;status:owned_no:found_no:stored_yes:has_offline_found_log_no:exclude_archived))
- - Stored in any list (no T4.5, T5): [- Stored in any list \(no T4.5, T5\):inconclusive=false:advanced=true]AND(stored_list:353:226:227:356:357:101:230:358:363:365:366:367:368:371:245:376:121:377:250:379:125:381:254:384:1:385:129:258:386:261:263:264:12:396:18:148:277:279:283:155:156:28:31:162:163:36:167:170:171:299:45:175:176:306:57:186:314:58:188:65:321:322:324:68:197:199:72:75:77:333:336:337:210:339:84:85:341:344:347:348:220:93:349:221:94:223:352;status:owned_no:found_no:stored_yes:has_offline_found_log_no:exclude_archived))
- Description:Leiter: [Description\:Leiter:inconclusive=false:advanced=false]AND(description:Leiter:contains))
- HQ + Jasmer + Matrix needed: [HQ + Jasmer + Matrix needed:inconclusive=false:advanced=true]AND(stored_list:129:162:356:36:358:365:337:371:148:245:250:283:155:384))
- Matrix Needed (List): [Matrix Needed \(List\):inconclusive=false:advanced=false]AND(stored_list:155))
- Matrix needed 4.5/4: [Matrix needed 4.5/4:inconclusive=false:advanced=true]AND(difficulty_terrain:d=4.5:d=4.5:t=4.0:t=4.0;status:owned_no:found_no:has_offline_found_log_no:exclude_archived))

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / PREMIUM
   extremcaching.com: Logged in (Login OK)
- Geocaching.com date format: dd MMM yy
- Geocaching.com website language: en-US
- Last login error on geocaching.com: communication error (2023-01-28 13-56)
- Last successful login on geocaching.com: 2023-02-12 12-01
- Routing: external / BRouter installed: true
- Installed c:geo plugins: contacts

Permissions
-------
-android.permission.ACCESS_COARSE_LOCATION:: granted
-android.permission.ACCESS_FINE_LOCATION:: granted
-android.permission.READ_CONTACTS:: DENIED
-android.permission.READ_EXTERNAL_STORAGE:: granted

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (48,4 GB free) v3 internal isDir(9 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (48,4 GB free) v3 internal isDir(9 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (48,4 GB free) v3 external non-removable isDir(5251 entries)
- Internal theme sync (is turned ON): /data/user/0/cgeo.geocaching/MapThemeData (48,4 GB free) v3 internal isDir(5 entries)
- Public Folders: #12
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=11, dirs:>=8, totalFileSize:>=2,0 GB, free space: 48,4 GB, files on device: 13278715)
  - OFFLINE_MAPS: Memory card/Locus/mapsVector (User-Defined)[Memory card/Locus/mapsVector[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector::]] (Uri: content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector/document/0123-4567%3ALocus%2FmapsVector, Av:true, files:>=58, dirs:>=6, totalFileSize:>=26,9 GB, free space: 125,3 GB, files on device: 0)
  - OFFLINE_MAP_THEMES: Memory card/Locus/mapsVector/_themes (User-Defined)[Memory card/Locus/mapsVector/_themes[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector%2F_themes::]] (Uri: content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector%2F_themes/document/0123-4567%3ALocus%2FmapsVector%2F_themes, Av:true, files:13, dirs:3, totalFileSize:5,5 MB, free space: 125,3 GB, files on device: 0)
  - OFFLINE_MAP_SHADING: Memory card/Locus/mapsVector/_hgt (Default)[Memory card/Locus/mapsVector/_hgt[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector::/_hgt]] (Uri: content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector/document/0123-4567%3ALocus%2FmapsVector%2F_hgt, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 125,3 GB, files on device: 0)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:2, dirs:0, totalFileSize:60,5 KB, free space: 48,4 GB, files on device: 13278715)
  - GPX: /Download/GPX (User-Defined)[/Download/GPX[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3ADownload%2FGPX::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ADownload%2FGPX/document/primary%3ADownload%2FGPX, Av:true, files:39, dirs:0, totalFileSize:5,3 MB, free space: 48,4 GB, files on device: 13278715)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:9, dirs:3, totalFileSize:2,0 GB, free space: 48,4 GB, files on device: 13278715)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:29, dirs:0, totalFileSize:168,8 KB, free space: 48,4 GB, files on device: 13278715)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:>=5, dirs:>=34, totalFileSize:>=1,4 MB, free space: 48,4 GB, files on device: 13278715)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:11, dirs:0, totalFileSize:109,9 KB, free space: 48,4 GB, files on device: 13278715)
  - ROUTING_TILES: Memory card/Android/media/btools.routingapp/brouter/segments4 (User-Defined)[Memory card/Android/media/btools.routingapp/brouter/segments4[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/0123-4567%3AAndroid%2Fmedia%2Fbtools.routingapp%2Fbrouter%2Fsegments4::]] (Uri: content://com.android.externalstorage.documents/tree/0123-4567%3AAndroid%2Fmedia%2Fbtools.routingapp%2Fbrouter%2Fsegments4/document/0123-4567%3AAndroid%2Fmedia%2Fbtools.routingapp%2Fbrouter%2Fsegments4, Av:true, files:20, dirs:0, totalFileSize:1,4 GB, free space: 125,3 GB, files on device: 0)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 48,4 GB, files on device: -1)
- Map render theme path: Elevate5_Locus/Elevate_Locus.zip:Elevate.xml
- PersistedDocumentUris: #1
- UNUSED_TRACK: null
- Persisted Uri Permissions: #5
  - content://com.android.externalstorage.documents/tree/0123-4567%3AAndroid%2Fmedia%2Fbtools.routingapp%2Fbrouter%2Fsegments4 (25 Dec 2021, 21:59):RW
  - content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector%2F_themes (22 Jan, 11:29):RW
  - content://com.android.externalstorage.documents/tree/primary%3ADownload%2FGPX (24 May 2021, 00:17):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (20 May 2021, 23:37):RW
  - content://com.android.externalstorage.documents/tree/0123-4567%3ALocus%2FmapsVector (20 May 2021, 23:37):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v99, Size:677,1 MB) on system internal storage

--- End of system information ---
```

### Additional Information

May be related to Cruiser changes, but may have been faulty for a long time already, as I have not used this setting for ages. 

Perhaps important, too: Cruiser is not yet installed (side note: was a bit astonished because of the difference in ratings between Android's Playstore (4.6 out of 120 reviews) and the rating shown on the website (3.9 out of 128 reviews) - which may of course just be a matter of actualization; but whoever is not up to date: either ratings have dropped dramatically in the younger past, or, even worse, a lot of critical reviews may have been filtered out or even deleted, which both does not really increase it's trustworthiness - thus I decided to stay off).

**Comment by Lineflyer** [comment]
was working for me in 2023.01.29-NB and after upgrading to todays nightly I can reproduce the crash.
Will try to get a log (problematic as as is not running well for me atm).

**Comment by Lineflyer** [comment]
It worked to get a log:

```
2023-02-12 12:58:45.463  4297-4297  Andro...time cgeo.geocaching  E  FATAL EXCEPTION: main
                                                                     Process: cgeo.geocaching, PID: 4297
                                                                     java.lang.NullPointerException: Attempt to invoke virtual method 'void androidx.preference.Preference.setSummary(int)' on a null object reference
                                                                     	at cgeo.geocaching.settings.fragments.PreferenceNavigationNavigationFragment.initNavigationMenuPreferences(PreferenceNavigationNavigationFragment.java:32)
                                                                     	at cgeo.geocaching.settings.fragments.PreferenceNavigationNavigationFragment.onCreatePreferences(PreferenceNavigationNavigationFragment.java:17)
                                                                     	at androidx.preference.PreferenceFragmentCompat.onCreate(PreferenceFragmentCompat.java:161)
                                                                     	at androidx.fragment.app.Fragment.performCreate(Fragment.java:2949)
                                                                     	at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:475)
                                                                     	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:278)
                                                                     	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
                                                                     	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
                                                                     	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
                                                                     	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
                                                                     	at android.os.Handler.handleCallback(Handler.java:942)
                                                                     	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                     	at android.os.Looper.loopOnce(Looper.java:226)
                                                                     	at android.os.Looper.loop(Looper.java:313)
                                                                     	at android.app.ActivityThread.main(ActivityThread.java:8741)
                                                                     	at java.lang.reflect.Method.invoke(Native Method)
                                                                     	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
                                                                     	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
2023-02-12 12:58:45.463  4297-4297  cgeo         cgeo.geocaching  W  [main] UncaughtException
                                                                     java.lang.NullPointerException: Attempt to invoke virtual method 'void androidx.preference.Preference.setSummary(int)' on a null object reference
                                                                     	at cgeo.geocaching.settings.fragments.PreferenceNavigationNavigationFragment.initNavigationMenuPreferences(PreferenceNavigationNavigationFragment.java:32)
                                                                     	at cgeo.geocaching.settings.fragments.PreferenceNavigationNavigationFragment.onCreatePreferences(PreferenceNavigationNavigationFragment.java:17)
                                                                     	at androidx.preference.PreferenceFragmentCompat.onCreate(PreferenceFragmentCompat.java:161)
                                                                     	at androidx.fragment.app.Fragment.performCreate(Fragment.java:2949)
                                                                     	at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:475)
                                                                     	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:278)
                                                                     	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
                                                                     	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
                                                                     	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
                                                                     	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
                                                                     	at android.os.Handler.handleCallback(Handler.java:942)
                                                                     	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                     	at android.os.Looper.loopOnce(Looper.java:226)
                                                                     	at android.os.Looper.loop(Looper.java:313)
                                                                     	at android.app.ActivityThread.main(ActivityThread.java:8741)
                                                                     	at java.lang.reflect.Method.invoke(Native Method)
                                                                     	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
                                                                     	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
2023-02-12 12:58:45.468  1467-4696  Activ...ager system_server    W  crash : cgeo.geocaching,10494
2023-02-12 12:58:45.469  1467-4696  Activ...ager system_server    W    Force finishing activity cgeo.geocaching/.settings.SettingsActivity
```

**Comment by MagpieFourtyTwo** [comment]
> (problematic as as is not running well for me atm).

If I can be of any help, just let me know (no matter if "not running well" refers to c:geo or an issue of your "general situation" ;) )

**Comment by Lineflyer** [comment]
@MagpieFourtyTwo 
Its about this problem: https://github.com/cgeo/cgeo/discussions/13956

**Comment by devemux86** [comment]
Cruiser 3.0.9 with the navigation communication with c:geo will be released soon.

> Perhaps important, too: Cruiser is not yet installed (side note: was a bit astonished because of the difference in ratings between Android's Playstore (4.6 out of 120 reviews) and the rating shown on the website (3.9 out of 128 reviews) - which may of course just be a matter of actualization; but whoever is not up to date: either ratings have dropped dramatically in the younger past, or, even worse, a lot of critical reviews may have been filtered out or even deleted, which both does not really increase it's trustworthiness - thus I decided to stay off).

"The Google Play Store will now provide device-specific ratings and reviews":

https://www.xda-developers.com/google-play-store-device-specific-ratings-and-reviews
https://www.androidpolice.com/play-store-ratings-relevant-devices

_"...ratings will be based on recent reviews and that reviews shown will be from those in the same region using the same type of device."_

**Comment by MagpieFourtyTwo** [comment]
Although it's off topic (and therefore we should end it here): I always deactivate all review filters and additionally sort by "most recent", cause these presets are not useful in whatever way and, worse, even distort the real user's rating result (qed).

In my eyes this "improvement", Google made for Android's PS ratings, is nothing else but just a mean try to improve overall ratings average, to let PS shine in a better light.  However, in the very end, it just turns out to be a cosmetic trick which ends up in misleading ratings, which then lead users to false or at least not reality based decisions - and this is bad - no matter what the reasons are (or what Google will make us believe of their honorable reasons).

**Comment by moving-bits** [comment]
> While trying to give Cruiser a try, I also tapped "Settings" => "Routing / Navigation" => "Navigation menu" for the first time since I guess years ... and it immediately crashed c:geo.

fixed with #13977
